### PR TITLE
Finished Environment 2021-2022 parsing

### PR DESCRIPTION
**Pull request by b47huang**
Still need to fix the case where the plan has two sets of courses to choose from

### CODE REVIEW

File: ProgramParsing/Environment/MajorParser2021_2022.py [hunk 1]
```diff
@@ -33,11 +33,14 @@ def _get_program(self):
...
         list = []
+        count_spaces = 0
         while i < len(info):
 
             line = info[i].strip().replace(" to ", "-")
-
-            if line == "": break
+            if line == "":
+                count_spaces += 1
+                if count_spaces == 3:
```
Comment (haowei99) [inline ProgramParsing/Environment/MajorParser2021_2022.py:46]: what is this used for again - why is it 3 now ?

Comment (haowei99) [reply ProgramParsing/Environment/MajorParser2021_2022.py:46]: and shouldn't you do else count_spaces=0 -resetting it if line!=""?

File: ProgramParsing/Environment/MajorParser2021_2022.py [hunk 2]
```diff
@@ -72,6 +75,52 @@ def _choose_x_course_list(self, i, info):
...
         return i, list
 
+    def _choose_option(self, i, info, program, relatedMajor, term):
+        option_num = 0
+        number_additional = 1
+        space_count = 0
+        while i < len(info):
+            line = info[i].strip().replace(" to ", "-")
+            # gets the option number
+            if line.split(" ")[0] == "Option":
```
Comment (haowei99) [inline ProgramParsing/Environment/MajorParser2021_2022.py]: line.split(" ")[0].lower() == "option"

File: ProgramParsing/Environment/MajorParser2021_2022.py [hunk 3]
```diff
@@ -72,6 +75,52 @@ def _choose_x_course_list(self, i, info):
...
+                option_num += 1
+                space_count = 0
+
+            # counts the empty lines
+            if line == "":
+                space_count += 1
+                if space_count == 5:
```
Comment (haowei99) [inline ProgramParsing/Environment/MajorParser2021_2022.py:97]: wait so you should continue after the line=="" ? This is for clarity - I don't think it changes anything

**Review by haowei99 — CHANGES_REQUESTED** [review]

File: ProgramParsing/Environment/MajorParser2021_2022.py (hunk 3 above)
Comment (b47huang) [reply ProgramParsing/Environment/MajorParser2021_2022.py:97]: The counting of "" is used to determine, when all the options have been looked at. I think the number of blank lines corresponds to the bullet indentation found on the pages.

**Review by haowei99 — APPROVED** [review]
i trust that u have run this already lol

**Comment by b47huang** [comment]
Don't worry, I've gone through the 9 plans like 50 times lol

**Pull request merged**